jwst/resample/resample_spec.py

```diff
@@ -157,15 +157,18 @@ def build_interpolated_output_wcs(self, refmodel=None):
                 ra_center_pt = np.nanmean(ra_center)
                 dec_center_pt = np.nanmean(dec_center)
 
-                # ra and dec this converted to tangent projection
-                tan = Pix2Sky_TAN()
-                native2celestial = RotateNative2Celestial(ra_center_pt, dec_center_pt, 180)
-                undist2sky1 = tan | native2celestial
-                # Filter out RuntimeWarnings due to computed NaNs in the WCS
-                warnings.simplefilter("ignore")
-                # at this center of slit find x,y tangent proction - x_tan, y_tan
-                x_tan, y_tan = undist2sky1.inverse(ra, dec)
-                warnings.resetwarnings()
+                if model.meta.wcs.output_frame.name == 'world':
+                    # ra and dec this converted to tangent projection
+                    tan = Pix2Sky_TAN()
+                    native2celestial = RotateNative2Celestial(ra_center_pt, dec_center_pt, 180)
+                    undist2sky1 = tan | native2celestial
+                    # Filter out RuntimeWarnings due to computed NaNs in the WCS
+                    warnings.simplefilter("ignore")
+                    # at this center of slit find x,y tangent proction - x_tan, y_tan
+                    x_tan, y_tan = undist2sky1.inverse(ra, dec)
+                    warnings.resetwarnings()
+                else:
+                    x_tan, y_tan = ra, dec
```

Comment:
Some commentary might be useful here to explain what's happening and why.

Reply:
Again, will comment to clarify, here and elsewhere